apps/src/weblab2/layout/VerticalLayout.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+import {FilePreview} from '@cdo/apps/codebridge/FilePreview';
+import {InfoPanel} from '@cdo/apps/codebridge/InfoPanel';
+import Workspace from '@cdo/apps/codebridge/Workspace';
+import {useVerticalLayout} from '@cdo/apps/lab2/hooks/useVerticalLayout';
+import ResizeBar from '@cdo/apps/lab2/views/components/layout/ResizeBar';
+
+import moduleStyles from '@cdo/apps/lab2/views/components/layout/layout.module.scss';
+
+const MIN_INFO_PANEL_WIDTH = 150;
+const INITIAL_INFO_PANEL_WIDTH = 300;
+const MIN_EDITOR_WIDTH = 300;
+const MIN_PREVIEW_WIDTH = 200;
+const INITIAL_PREVIEW_WIDTH = 600;
+
+const VerticalLayout: React.FunctionComponent = () => {
+  const {
+    leftPanelWidth,
+    middlePanelWidth,
+    rightPanelWidth,
+    leftPanelSeparatorProps,
+    leftPanelDragging,
+  } = useVerticalLayout({
+    leftPanel: {
+      minWidth: MIN_INFO_PANEL_WIDTH,
+      initialWidth: INITIAL_INFO_PANEL_WIDTH,
+      name: 'instructions',
+    },
+    middlePanel: {
+      minWidth: MIN_EDITOR_WIDTH,
+      name: 'editor',
+    },
+    rightPanel: {
+      minWidth: MIN_PREVIEW_WIDTH,
+      initialWidth: INITIAL_PREVIEW_WIDTH,
+      name: 'preview',
+    },
+  });
+
+  return (
+    <div className={moduleStyles.layoutContainer}>
+      <InfoPanel
+        style={{width: leftPanelWidth}}
+        className={moduleStyles.flexShrink0}
+      />
+      <ResizeBar
+        isVertical={true}
+        separatorProps={leftPanelSeparatorProps}
+        isDragging={leftPanelDragging}
+      />
+      <Workspace
+        style={{width: middlePanelWidth}}
+        className={moduleStyles.shrinkAndGrow}
+      />
+      {/* TODO: Make right panel resizable. The iframe in FilePreview makes it so you
+         can only drag left, not right (something about the mouse events getting 
+         captured by the preview?) */}
```

Comment:
Just noting that in the horizontal layout of weblab, it looks like this is also an issue if the mouse is on the resize bar between the preview and instructions.

https://github.com/user-attachments/assets/1e7e2d7d-9cac-4cc1-a491-605dabc0ff1b

Reply:
Good call out. This is the problem I was seeing with vertical mode. I think I'll file a ticket for this and leave as-is? Since no one is using Web lab 2. Or do you think it's egregious enough that I should remove that resize bar too?

Reply:
I think that's a good call. Now noticing that there's another unexpected thing happening when trying to resize between preview and instructions. You end up being able to move the resize bar without being clicked on (on hover). 

https://github.com/user-attachments/assets/5cb24bda-9391-4bc3-b99f-5fdf2106b137

Reply:
yeah the preview makes things really buggy